Backend/src/controllers/user.controller.ts

```diff
@@ -108,9 +109,25 @@ export const uploadFile = async (req: Request, res: Response) => {
       last_claimed_slot: 1,
     };
     await db.insert(depositAccount).values(depositItem).returning();
+    const depositIx = await createDepositInstruction(
+      publicKey,
+      uploadObject.cid,
+      uploadObject.size,
+      durationInSeconds,
+      depositItem.deposit_amount
+    );
     res.status(200).json({
       message: "Successfully uploaded the object",
       object: uploadObject,
+      instruction: {
+        programId: depositIx.programId.toBase58(),
+        keys: depositIx.keys.map((key) => ({
+          pubkey: key.pubkey.toBase58(),
+          isSigner: key.isSigner,
+          isWritable: key.isWritable,
+        })),
+        data: depositIx.data.toString("base64"),
+      },
```

Comment:
*suggestion*: instead of constructing the deposit instruction here, you should import the `createDepositTransaction` function that already does it in the Solana controller, and normalize its response.